src/fftvis/cpu/cpu_simulate.py

```diff
@@ -472,48 +478,48 @@ def _evaluate_vis_chunk(
                         interpolation_function=interpolation_function,
                     ).astype(complex_dtype)
 
-                    if polarized:
+                    if polarized and polarized_sky_model:
                         _cpu_beam_evaluator.get_apparent_flux_polarized(
-                            A_s, flux[:nsim_sources, freqidx] if flux.ndim > 1 else flux[:nsim_sources]
+                            apparent_coherency, np.transpose(flux[:nsim_sources, freqidx], (1, 2, 0))
+                        )
+                    elif polarized:
+                        _cpu_beam_evaluator.get_apparent_flux_polarized_beam(
+                            apparent_coherency, flux[:nsim_sources, freqidx]
                         )
                     else:
-                        # Check if flux is 1D or 2D
-                        if flux.ndim > 1:
-                            A_s *= flux[:nsim_sources, freqidx]
-                        else:
-                            A_s *= flux[:nsim_sources]
+                        apparent_coherency *= flux[:nsim_sources, freqidx]
+
 
                     # Check if A_s can be reshaped to expected dimensions
                     # For polarized case with 2 feeds, expected shape is (2, 2, nsim_sources) -> (4, nsim_sources)
                     expected_size = nfeeds**2 * nsim_sources
-                    if A_s.size != expected_size: # pragma: no cover
+                    if apparent_coherency.size != expected_size: # pragma: no cover
                         # Log the shape mismatch and try to adapt
-                        logger.warning(f"Shape mismatch: A_s size {A_s.size} != expected size {expected_size}") # pragma: no cover
-                        logger.warning(f"A_s shape: {A_s.shape}, nfeeds: {nfeeds}, nsim_sources: {nsim_sources}") # pragma: no cover
+                        logger.warning(f"Shape mismatch: A_s size {apparent_coherency.size} != expected size {expected_size}") # pragma: no cover
+                        logger.warning(f"A_s shape: {apparent_coherency.shape}, nfeeds: {nfeeds}, nsim_sources: {nsim_sources}") # pragma: no cover
                         
                         # Handle polarized case specially - if we got a 2D array but expected 3D
-                        if polarized and A_s.ndim == 2: # pragma: no cover
+                        if polarized and apparent_coherency.ndim == 2: # pragma: no cover
```

Comment:
Why don't we raise an error in this case?

Reply:
I just removed this check.